src/main/java/com/scaleunlimited/flinkcrawler/warc/CreateWARCWritableFunction.java

```diff
@@ -0,0 +1,111 @@
+package com.scaleunlimited.flinkcrawler.warc;
+
+import java.io.ByteArrayInputStream;
+import java.io.DataInputStream;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+import org.apache.commons.io.output.ByteArrayOutputStream;
+import org.apache.flink.api.java.tuple.Tuple2;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.util.Collector;
+import org.apache.hadoop.io.NullWritable;
+
+import com.scaleunlimited.flinkcrawler.functions.BaseFlatMapFunction;
+import com.scaleunlimited.flinkcrawler.pojos.FetchResultUrl;
+import com.scaleunlimited.flinkcrawler.pojos.FetchStatus;
+
+@SuppressWarnings("serial")
+public class CreateWARCWritableFunction
+        extends BaseFlatMapFunction<FetchResultUrl, Tuple2<NullWritable, WARCWritable>> {
+
+    private static String SOFTWARE = "flink-crawler";
+
+    private SimpleDateFormat _dateFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'");
+    private String _userAgentString;
+    private boolean _warcInfoEmitted;
+
+    public CreateWARCWritableFunction(String userAgentString) {
+        _userAgentString = userAgentString;
+    }
+
+    @Override
+    public void open(Configuration parameters) throws Exception {
+        super.open(parameters);
+        _warcInfoEmitted = false;
+    }
+
+    @Override
+    public void flatMap(FetchResultUrl fetchResultUrl,
+            Collector<Tuple2<NullWritable, WARCWritable>> collector) throws Exception {
+
+        if (fetchResultUrl.getStatus() != FetchStatus.FETCHED) {
+            return;
+        }
+
+        if (!_warcInfoEmitted) {
+            outputWARCInfoRecord(collector);
+            _warcInfoEmitted = true;
+        }
+
+        outputWARCResourceRecord(collector, fetchResultUrl);
+    }
+
+    private void outputWARCInfoRecord(Collector<Tuple2<NullWritable, WARCWritable>> collector)
+            throws IOException {
+    
+        StringBuffer content = new StringBuffer();
+    
+        content.append(String.format("software: %s\r\n", SOFTWARE));
```

Comment:
No big deal, but I read recently that as of Java 5 the concatenation operator (+) gets compiled into the same byte codes that are used for StringBuffer() - See https://stackoverflow.com/questions/44117788/performance-between-string-format-and-stringbuilder?utm_medium=organic&utm_source=google_rich_qa&utm_campaign=google_rich_qa